include/rmm/mr/device/block_aligned_resource_adaptor.hpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <optional>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/detail/aligned.hpp>
+#include <rmm/detail/error.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+namespace rmm {
+namespace mr {
+/**
+ * @brief Resource that adapts `Upstream` memory resource to allocate memory in multiples of
+ * filesystem block size (default to 4096).
+ *
+ * An instance of this resource can be constructed with an existing, upstream resource in order
+ * to satisfy allocation requests. This adaptor wraps allocations and deallocations from Upstream
+ * with block size alignment.
+ *
+ * @tparam Upstream Type of the upstream resource used for allocation/deallocation.
+ */
+template <typename Upstream>
+class block_aligned_resource_adaptor final : public device_memory_resource {
```

Comment:
Can we just call this `aligned_resource_adaptor`? I don't think there's any need to refer to "blocks" or filesystems.

Reply:
+1

Reply:
Done.